src/backend/db/database_manager.py

```diff
@@ -259,3 +296,128 @@ def check_customer_guid_exists(self, customer_guid):
         finally:
             session.close()
 
+    #Custom Fields Related Functions
+    def add_custom_field(self, customer_guid, field_name, field_type, required):
+        customer_db_name = self.get_customer_db(customer_guid)
+        session = DatabaseManager._session_factory()
+
+        # Define a set of allowed SQL data types
+        allowed_sql_types = {"VARCHAR(255)", "INT", "BOOLEAN", "DATE", "MEDIUMTEXT"}
+
+        # Validate the field type
+        if field_type not in allowed_sql_types:
+            raise ValueError(f"Unsupported field type: {field_type}. Allowed types are: {', '.join(allowed_sql_types)}")
+
+        try:
+            logger.debug(f"Switching to customer database: {customer_db_name}")
+            session.execute(text(f"USE `{customer_db_name}`"))
+
+            # Add metadata to custom_fields table
+            query = '''INSERT INTO custom_fields (field_name, field_type, required)
+                       VALUES (:field_name, :field_type, :required)'''
+            session.execute(
+                text(query),
+                {
+                    "field_name": field_name,
+                    "field_type": field_type,
+                    "required": required,
+                },
+            )
+
+            # Dynamically add column to custom_field_values table
+            alter_table_query = f"""
+            ALTER TABLE custom_field_values
+            ADD COLUMN {field_name} {field_type};
```

Comment:
If there are rows present in the table, make sure the column added contains NULL values

like 

ALTER TABLE custom_field_values
            ADD COLUMN {field_name} {field_type}; with value as NULL 
            
            for all the pre-existing rows

Reply:
sir by default it add the new column with null values